blockchain/blocks/src/tipset.rs

```diff
@@ -429,4 +445,68 @@ mod test {
         // Can not break weight tie with all min tickets the same
         assert!(!ts6.break_weight_tie(&ts7));
     }
+
+    #[test]
+    fn ensure_miner_addresses_are_distinct() {
+        let h0 = BlockHeader::builder()
+            .miner_address(Address::new_id(0))
+            .build()
+            .unwrap();
+        let h1 = BlockHeader::builder()
+            .miner_address(Address::new_id(0))
+            .build()
+            .unwrap();
+        assert!(Tipset::new(vec![h0, h1]).unwrap_err() == Error::InvalidTipset("miner_addresses are not distinct".to_string()));
+    }
```

Comment:
Good. Could you also add a test that would have caught the error in the previous code? Ie, headers that are not distinct but still different from the first header.